src/Symfony/Component/HttpClient/Response/MockResponse.php

```diff
@@ -204,7 +204,7 @@ protected static function select(ClientState $multi, float $timeout): int
     private static function writeRequest(self $response, array $options, ResponseInterface $mock)
     {
         $onProgress = $options['on_progress'] ?? static function () {};
-        $response->info += $mock->getInfo() ?: [];
+        $response->info += $mock->getInfo();
```

Comment:
`getInfo()` must return an array by contract if no `$type` is provided, I think we only needed that fallback because of bad mocks in tests so I removed it.

Reply:
We'd better keep it to not break anything, we don't care if mocks are not fully compliant